custom_db_role resource use *string

Freyert opened the pull request:
## Description

Fixes https://github.com/mongodb/go-client-mongodb-atlas/issues/263

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

## Further comments

Just posting this here to streamline any fix on this. If further doc/tests/etc. are needed let me know. Didn't want to run the tests in case they were integration and not just unit tests.

I did use a `replace` in the `go.mod` to use my fork and this fixes my issue, but you have to wrap all strings as pointers. Therefore, this is a breaking change.

themantissa reviewed (dismissed):
LGTM.  Added DoU so one can review so we can figure out the impact downstream.  TY!

Freyert commented:
I reworked the test to loop over a list of cases so it is easier to add more cases. I also added an extra case to ensure when we use "cluster" resources it works correctly.

Freyert commented on mongodbatlas/custom_db_roles.go:
```diff
@@ -41,9 +41,9 @@ var _ CustomDBRolesService = &CustomDBRolesServiceOp{}
-	Db         string `json:"db,omitempty"` //nolint:stylecheck // not changing this as is a breaking change
-	Cluster    *bool  `json:"cluster,omitempty"`
+	Collection *string `json:"collection,omitempty"`
+	Db         *string `json:"db,omitempty"` //nolint:stylecheck // not changing this as is a breaking change
```
Just added this eb1bc7a (misunderstood it the first time)

Freyert commented on mongodbatlas/custom_db_roles_test.go:
```diff
@@ -42,8 +42,8 @@ func TestCustomDBRoles_ListCustomDBRoles(t *testing.T) {
 			Resources: []Resource{{
-				Collection: "test-collection",
-				Db:         "test-db",
+				Collection: StrPtr("test-collection"),
```
dfa4844

gssbzn reviewed (approved):
LGTM thanks for all the changes here

coderGo93 reviewed (approved):
LGTM

themantissa commented:
@gssbzn  we are okay for this to be merged.  We'll address it in the provider.
